evm/src/byte_packing/byte_packing_stark.rs

```diff
@@ -194,34 +205,28 @@ impl<F: RichField + Extendable<D>, const D: usize> BytePackingStark<F, D> {
             virt,
         } = base_address;
 
-        let mut rows = Vec::with_capacity(bytes.len());
         let mut row = [F::ZERO; NUM_COLUMNS];
         row[IS_READ] = F::from_bool(is_read);
 
         row[ADDR_CONTEXT] = F::from_canonical_usize(context);
         row[ADDR_SEGMENT] = F::from_canonical_usize(segment);
-        // Because of the endianness, we start by the final virtual address value
-        // and decrement it at each step. Similarly, we process the byte sequence
-        // in reverse order.
-        row[ADDR_VIRTUAL] = F::from_canonical_usize(virt + bytes.len() - 1);
+        // We store the initial virtual segment. But the CTLs,
+        // we start with virt + sequence_len - 1.
+        row[ADDR_VIRTUAL] = F::from_canonical_usize(virt);
 
         row[TIMESTAMP] = F::from_canonical_usize(timestamp);
 
-        for (i, &byte) in bytes.iter().rev().enumerate() {
-            if i == bytes.len() - 1 {
-                row[SEQUENCE_END] = F::ONE;
-            }
-            row[value_bytes(i)] = F::from_canonical_u8(byte);
-            row[index_bytes(i)] = F::ONE;
+        if !bytes.is_empty() {
+            row[index_len(bytes.len() - 1)] = F::ONE;
 
-            rows.push(row);
+            for (i, &byte) in bytes.iter().rev().enumerate() {
+                row[value_bytes(i)] = F::from_canonical_u8(byte);
+            }
 
-            // Update those fields for the next row
-            row[index_bytes(i)] = F::ZERO;
-            row[ADDR_VIRTUAL] -= F::ONE;
+            Some(row)
+        } else {
+            None
         }
-
-        rows
```

Comment:
Why doing all of this if the sequence is empty and we return `None` in the end?
We should probably just filter ahead empty sequences so that this always returns a row (for non-empty ones).